src/tests/test_update.py

```diff
@@ -102,4 +105,71 @@ def test_statistics(self, mock):
         self.assertEqual(results['num_recompares'], 1)
         self.assertEqual(results['num_stored_events'], 1)
         self.assertEqual(results['num_skipped_events_skipped'], 1)
-        self.assertEqual(results['num_applied_applied'], 1)
\ No newline at end of file
+        self.assertEqual(results['num_applied_applied'], 1)
+
+    def test_gob_event_action(self, mock_event):
+        # setup event state
+        event = {
+            'sa_instance_state': '',
+            'version': '0.1',
+            'catalogue': 'test_catalogue',
+            'eventid': 20,
+            'application': 'TEST',
+            'source_id': 'id 2',
+            'action': None,
+            'entity': 'test_entity',
+            'timestamp': datetime.datetime(2019, 1, 30, 18, 7, 7),
+            'contents': '{"_last_event": 18, "_source_id": "id 2", "_entity_source_id": "id 2"}',
+            'source': 'test'
+        }
+
+        # setup data
+        data = {'_last_event': 1, '_source_id': 'id 2', '_entity_source_id': 'id 2'}
+
+        last_event_expected = 1
+        for action_expected in ['ADD', 'DELETE', 'CONFIRM', 'MODIFY']:
+            data['_last_event'] = last_event_expected
+            event['action'] = action_expected
+
+            # from dictionary to object with attributes
+            dummy_event = namedtuple("mockEvent", event.keys())(*event.values())
+
+            # setup done, run gob event
+            gob_event = _get_gob_event(dummy_event, data)
+
+            # assert action and lasst_event are as expected
+            self.assertEqual(action_expected, gob_event.name)
+            self.assertEqual(last_event_expected, gob_event.last_event)
+
+            last_event_expected += 1
+
+    def test_gob_event_invalid_action(self, mock_event):
+        # setup event state
+        event = {
+            'sa_instance_state': '',
+            'version': '0.1',
+            'catalogue': 'test_catalogue',
+            'eventid': 20,
+            'application': 'TEST',
+            'source_id': 'id 2',
+            'action': None,
+            'entity': 'test_entity',
+            'timestamp': datetime.datetime(2019, 1, 30, 18, 7, 7),
+            'contents': '{"_last_event": 18, "_source_id": "id 2", "_entity_source_id": "id 2"}',
+            'source': 'test'
+        }
+
+        # setup data
+        data = {'_last_event': 1, '_source_id': 'id 2', '_entity_source_id': 'id 2'}
+
```

Comment:
event and data are shared. They can be encapsulated in a fixture or a test method

Reply:
The construct to convert to a namedtuple can then be part of the test method